src/main/java/com/basho/riak/client/core/query/indexes/RiakIndex.java

```diff
@@ -352,7 +352,7 @@ public String toString()
      * @param <T> the RiakIndex this Name encapsulates
      * @see RiakIndexes
      */
-    public static abstract class Name<T extends RiakIndex>
+    public static abstract class Name<T extends RiakIndex<?>>
```

Comment:
Yeah, it says that the type parameter used by RiakIndex will be of some specific (but unknown) type.  We don't use the type parameter wrapped by RiakIndex anywhere in the Name class, so we _could_ use either. On the flip side I don't see how you could ever use the raw type by itself (`RiakIndex`, no type parameter), since RiakIndex is an abstract class.